1-Draft/RFCNNNN-VSCode-PSScriptAnalyzer-Rules.md

```diff
@@ -0,0 +1,151 @@
+| AvoidUsingUsernameAndPasswordParams          | Error       | Aggressive |                |
+| AvoidUsingWMICmdlet                          | Warning     | Aggressive | x              |
+| AvoidUsingWriteHost                          | Warning     | Aggressive |                |
+| DSCDscExamplesPresent                        | Information |            |                |
+| DSCDscTestsPresent                           | Information |            |                |
+| DSCReturnCorrectTypesForDSCFunctions         | Information |            |                |
+| DSCStandardDSCFunctionsInResource            | Error       |            |                |
+| DSCUseIdenticalMandatoryParametersForDSC     | Error       |            |                |
+| DSCUseIdenticalParametersForDSC              | Error       |            |                |
+| DSCUseVerboseMessageInDSCResource            | Error       |            |                |
+| MisleadingBacktick                           | Warning     | Default    |                |
+| MissingModuleManifestField                   | Warning     | Default    |                |
+| PossibleIncorrectComparisonWithNull          | Warning     | Default    |                |
+| PossibleIncorrectUsageOfAssignmentOperator   | Warning     | Aggressive |                |
+| PossibleIncorrectUsageOfRedirectionOperator  | Warning     | Default    |                |
+| ProvideCommentHelp                           | Information |            |                |
+| ReservedCmdletChar                           | Error       | Default    |                |
+| ReservedParams                               | Error       | Default    |                |
+| ShouldProcess                                | Error       | Default    |                |
+| UseApprovedVerbs                             | Warning     | Default    |                |
+| UseBOMForUnicodeEncodedFile                  | Warning     |            |                |
+| UseCmdletCorrectly                           | Warning     | Aggressive |                |
+| UseDeclaredVarsMoreThanAssignments           | Warning     | Default    |                |
+| UseLiteralInitializerForHashtable            | Warning     | Aggressive |                |
+| UseOutputTypeCorrectly                       | Information | Aggressive |                |
+| UsePSCredentialType                          | Warning     | Aggressive |                |
+| UseShouldProcessForStateChangingFunctions    | Warning     | Aggressive |                |
+| UseSingularNouns                             | Warning     | Aggressive | x              |
+| UseSupportsShouldProcess                     | Warning     | Aggressive | x              |
+| UseToExportFieldsInManifest                  | Warning     | Default    |                |
+| UseCompatibleCmdlets                         | Warning     |            |                |
```

Comment:
Correct decision because the rule does nothing by default. In the future we could tweak it to use the current ps version/platform but the rule is not ready for that yet.